src/rational/mat_q/gso.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of the Gram-Schmidt Orthogonalization for ['MatQ'] matrices.
+
+use super::MatQ;
+use crate::traits::{GetNumColumns, GetNumRows};
+use flint_sys::fmpq_mat::fmpq_mat_gso;
+
+impl MatQ {
+    /// Computes the Gram-Schmidt Orthogonalization of the matrix and returns a [`MatQ`] with the corresponding matrix.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use std::str::FromStr;
+    ///
+    /// let mat = MatQ::from_str("[[1/2, 1],[2, 5]]").unwrap();
+    /// let mat_gso = mat.gso();
+    /// ```
+    pub fn gso(&self) -> Self {
+        let mut out = MatQ::new(self.get_num_rows(), self.get_num_columns()).unwrap();
+        unsafe {
+            fmpq_mat_gso(&mut out.matrix, &self.matrix);
+        }
+        out
+    }
+}
+
+#[cfg(test)]
+mod test_gso {
+
```

Comment:
please add tests for different dimensions.
What happens if some columns or rows are identical? (linear dependent)
Please add a test where some entries are large.
Please add special cases: such as a matrix with only one column, one row, etc.
What if we have more columns than rows?